NGit/NGit.Patch/FileHeader.cs

```diff
@@ -711,6 +715,10 @@ private static string P1(string r)
 		internal virtual FileMode ParseFileMode(int ptr, int end)
 		{
 			int tmp = 0;
+		    if (buf[end - 2] == '\r') // Ignore windows line ending
```

Comment:
Its not a check for `\r\r\n` but I'll go have a shot at pulling out a well named method.